apps/devhub/tests/test_utils.py

```diff
@@ -0,0 +1,373 @@
+import os.path
+from copy import deepcopy
+
+import mock
+import pytest
+from mock import call
+
+from django.conf import settings
+
+import amo
+import amo.tests
+from addons.models import Addon
+from files.models import File, FileUpload
+from versions.models import Version
+from devhub import utils
+from devhub.tasks import annotate_validation_results
+
+
+pytestmark = pytest.mark.django_db
+
+
+def change_dict(src, changes):
+    src = src.copy()
+    src.update(changes)
+    return src
+
+
+class TestValidationComparator(amo.tests.TestCase):
+    SIGNING_SUMMARY = {'high': 0, 'medium': 0, 'low': 0, 'trivial': 0}
+
+    def compare(self, old, new, expected):
+        """Compare two messages, and assert that the expected annotations are
+        present.
+
+        `old` is a message dict from the previous set of results,
+        `new` is a dict containing the set of property changes between
+        the old message and the new one, and `expected` is the full set of
+        annotations expected to be added to the new message."""
+
+        new = change_dict(old, new)
+        expected = change_dict(new, expected)
+
+        if 'matched' in expected:
+            # Special case: If we expect a match, the value should
+            # be the original message dict, so insert that value.
```

Comment:
I wrote it this way 1) for symmetry between the list of changes introduced in the newer validation, and the list of changes introduced by the annotation, and 2) with the expectation that we'd be introducing new annotations in the future. I'll try to think about cleaner ways we might be able to do this.